### PR TITLE
Release/v0.0.30

### PR DESCRIPTION
**Pull request by jasonbahl**
# Release Notes

## User Role Queries
- You can now query for user role info using GraphQL!
- The following rootQueries now exist: `userRole` and `userRoles`
- example: 
```
query GetUserRoles {
  userRoles {
    edges {
       node {
           name
           id
        }
     }
  }
}
```

## Comment Mutations
- You can now mutate comments! 
- The following root mutations are now available: `createComment`, `deleteComment`, `updateComment`, `restoreComment`

## Menu Queries
- You can now query for menus (and menuItems)!
- The following root queries exist now: `menu`, `menus`, `menuItem`, `menuItems` 
- example: 
```
query GetMenuItems {
  menuItems {
    edges { 
       node {
          menuItemId
          connectedObject {
             ... on Post {
                title
                postId
             }
             ... on Category {
                name
                categoryId
             }
          }
       }
    }
  }
}
```

## New Docker Environment!
- You can now easily launch a Docker environment for developing with/for WPGraphQL. At the root of the plugin is a `docker-compose.dev.yml` file with further instructions outlined on usage. 

### Misc changes / bug fixes
- update `.gitignore` to ignore `.vscode` files and `phpunit.xml`
- update `bin/install-wp-tests.sh` to allow for overriding the `$DB_USER`when setting up tests
- fixed a bug with `$request` variables not working right in one of the hooks
- fixed a bug with `apply_filters( 'the_title' )` being applied incorrectly
- WPUnionType introduced (in favor of GraphQL/UnionType)
- fixed bug with `sourceUrl` not properly being returned when querying for `sizes` on a `mediaItem`
- made `graphql_init` a pluggable function allowing for it to be initialized externally once only for themes/plugins that include it in their codebase
--------

# BREAKING CHANGES:
- removed filter `graphql_type_config` in favor of `graphql_wp_object_type_config`
- removed filter `graphql_comment_author_union_possible_types` in favor of the centralized filter `graphql_wp_union_type_config` in the new WPUnionType
- removed filter `graphql_post_object_union_possible_types` in favor of the centralized filter `graphql_wp_union_type_config` in the new WPUnionType
- removed filter `graphql_term_object_union_possible_types` in favor of the centralized filter `graphql_wp_union_type_config` in the new WPUnionType

--------

Thanks for contributing to this release @chriszarate @CodeProKid @ElisaMassafra @timhanlon @natewoodbridge @jmlallier @kidunot89!

### CODE REVIEW

File: src/Data/DataSource.php [hunk 1]
```diff
@@ -371,6 +372,44 @@ public static function resolve_users_connection( $source, array $args, $context,
...
+	 * @since  0.0.30
+	 * @access public
+	 */
+	public static function resolve_user_role( $name ) {
+
+		$role = get_role( $name );
+
+		if ( null === $role ) {
+			throw new UserError( sprintf( __( 'No user role was found with the name %s', 'wp-graphql' ), $name ) );
+		} else {
```
Comment (wolzogen) [inline src/Data/DataSource.php]: You can rewrite:
```php
if ( null === $role || !is_array($role) || !$role) {
    throw new UserError( sprintf( __( 'No user role was found with the name %s', 'wp-graphql' ), $name ) );
}
$role['id'] = $name;
return $role;
```

File: src/Type/MenuItem/Connection/MenuItemConnectionDefinition.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+			
+			self::$where_args = new WPInputObjectType( [
+				'name'   => 'MenuItemQueryArgs',
+				'fields' => function() {
+					return self::where_fields();
+				},
+			] );
+		}
+
+		return ! empty( self::$where_args ) ? self::$where_args : null;
```
Comment (wolzogen) [inline src/Type/MenuItem/Connection/MenuItemConnectionDefinition.php]: See [PHP type comparison tables](http://php.net/manual/en/types.comparisons.php)
```php
return self::$where_args ?: null;
```

**Comment by jasonbahl** [comment]
@wolzogen I saw some comments come through from you, but I don't see them anymore. 🤔 

Did you delete them or is Github having some issues?

**Comment by jasonbahl** [comment]
@wolzogen if you feel your comments (that I can't see anymore) are valid, can you open issues and/or PRs and we can discuss further if needed. 

Thanks!

**Pull request merged**